### ISSUE TITLE
v0.2.1 release mainnet.zip has a testnet configuration

### ISSUE THREAD

**Issue by ia03** [0xPolygonHermez/zkevm-node#2369]
#### System information

zkEVM Node version: `v0.2.1`
OS & Version: `Linux`
Commit hash : `N/A`
Network: `Mainnet`

#### Expected behaviour
`mainnet.zip` is configured for mainnet

#### Actual behaviour
`mainnet.zip` is configured for testnet. It has a config/environments/testnet folder and `docker-compose.yml` references testnet

#### Steps to reproduce the behaviour
Download `mainnet.zip` at https://github.com/0xPolygonHermez/zkevm-node/releases/tag/v0.2.1

#### Backtrace

````
N/A
````

When submitting logs: please submit them as text and not screenshots.

**Comment by KonScanner** [comment]
Even when you change:
```
 volumes:
      - ${ZKEVM_ADVANCED_CONFIG_DIR:-./config/environments/testnet}/testnet.node.config.toml:/app/config.toml
```
to:
```
 volumes:
      - ${ZKEVM_ADVANCED_CONFIG_DIR:-./config/environments/testnet}/public.node.config.toml:/app/config.toml
```

Everywhere where there's the issue I still get the following failures:

```
2023-08-03T19:04:38.107Z        FATAL   cmd/main.go:193 Required flag "cfg" not set
/src/log/log.go:140 github.com/0xPolygonHermez/zkevm-node/log.appendStackTraceMaybeArgs()
/src/log/log.go:221 github.com/0xPolygonHermez/zkevm-node/log.Fatal()
/src/cmd/main.go:193 main.main()
/usr/local/go/src/runtime/proc.go:250 runtime.main()
        {"pid": 1, "version": "v0.2.1"}
main.main
        /src/cmd/main.go:193
runtime.main
        /usr/local/go/src/runtime/proc.go:250
```
On both of:
![image](https://github.com/0xPolygonHermez/zkevm-node/assets/42744293/c797c0d7-20d8-4139-b6a7-743dcb4a69a8)

**Comment by ToniRamirezM** [comment]
Please, check this issue:

https://github.com/0xPolygonHermez/zkevm-node/issues/2346

And the one linked there.

**Comment by KonScanner** [comment]
Pre-release has this issue now:
```
2023-08-04 20:59:40.596 GMT [1] LOG:  input in flex scanner failed at file "/etc/postgresql.conf" line 1
2023-08-04 20:59:40.596 GMT [1] FATAL:  configuration file "/etc/postgresql.conf" contains errors
```
On the `state-db` postgres
![image](https://github.com/0xPolygonHermez/zkevm-node/assets/42744293/e1092ed4-dc57-4fce-97a8-91826a92e491)

**Comment by rmaganza** [comment]
> Pre-release has this issue now:
> 
> ```
> 2023-08-04 20:59:40.596 GMT [1] LOG:  input in flex scanner failed at file "/etc/postgresql.conf" line 1
> 2023-08-04 20:59:40.596 GMT [1] FATAL:  configuration file "/etc/postgresql.conf" contains errors
> ```
> 
> On the `state-db` postgres ![image](https://user-images.githubusercontent.com/42744293/258530454-e1092ed4-dc57-4fce-97a8-91826a92e491.png)

Second this, unable to run on latest release or pre-release.

**Comment by ToniRamirezM** [comment]
I think this may be related to the config file being prepared for postgres 14 and the docker images not being fixed.

Could you please fix the version of every postgres image to 14?

`image: postgres:14`

**Comment by ToniRamirezM** [comment]
Can you please provide your postgres version?

**Comment by rmaganza** [comment]
I tried with the latest release (0.3.0) and I still have the same postgres error. I'm using the Postgres version specificied in the original docker-compose file (15).

**Comment by abhijeetbhagat** [comment]
> Pre-release has this issue now:
> 
> ```
> 2023-08-04 20:59:40.596 GMT [1] LOG:  input in flex scanner failed at file "/etc/postgresql.conf" line 1
> 2023-08-04 20:59:40.596 GMT [1] FATAL:  configuration file "/etc/postgresql.conf" contains errors
> ```
> 
> On the `state-db` postgres ![image](https://user-images.githubusercontent.com/42744293/258530454-e1092ed4-dc57-4fce-97a8-91826a92e491.png)

i am facing the same problem and would like to fix it myself provided there's some guidance.

**Comment by bimlas** [comment]
I think it's an issue with `docker-compose`, because nested parameter expansion is supported since compose v2.2.0 (https://github.com/docker/compose/issues/7794). 

If you get the same output (`${ZKEVM_NETWORK}`), you have to upgrade `docker-compose`:

```
ls app/mainnet/config/environments/
'${ZKEVM_NETWORK}'   mainnet   public   testnet
```

Please check your `docker-compose --version`.

**Comment by rmaganza** [comment]
I'm facing the issue and I have docker-compose v2.21

**Comment by bimlas** [comment]
I just replaced `${ZKEVM_NETWORK}` by `mainnet` in `app/mainnet/docker-compose.yml`, now it works.

**EDIT**

zkEVM version 0.3.1

**Comment by rmaganza** [comment]
I can confirm both the network name and the postgres issue are fixed in the latest pre-release (v0.3.1)
